src/HealthChecks.CosmosDb/DependencyInjection/CosmosDbHealthCheckBuilderExtensions.cs

```diff
@@ -1,245 +1,394 @@
+        /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+        /// <returns>The specified <paramref name="builder"/>.</returns>
+        public static IHealthChecksBuilder AddAzureTable(
+            this IHealthChecksBuilder builder,
+            Uri endpoint,
+            TableSharedKeyCredential credentials,
+            string? tableName,
+            string? name = default,
+            HealthStatus? failureStatus = default,
+            IEnumerable<string>? tags = default,
+            TimeSpan? timeout = default)
+        {
+            return builder.Add(new HealthCheckRegistration(
+               name ?? TABLE_NAME,
+               sp => new TableServiceHealthCheck(endpoint, credentials, tableName),
+               failureStatus,
+               tags,
+               timeout));
+        }
+
+        /// <summary>
+        /// Add a health check for Azure CosmosDb/Azure Storage table.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+        /// <param name="endpoint">The CosmosDB Table or Azure Storage Table uri endpoint.</param>
+        /// <param name="tokenCredential">An instance of <see cref="TokenCredential"/> to be used for authentication</param>
+        /// <param name="tableName">Table name to check for existence.</param>
+        /// <param name="name">The health check name. Optional. If <see langword="null"/> the type name 'cosmosdb' will be used for the name.</param>
+        /// <param name="failureStatus">
+        /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <see langword="null"/> then
+        /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+        /// </param>
+        /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+        /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+        /// <returns>The specified <paramref name="builder"/>.</returns>
+        public static IHealthChecksBuilder AddAzureTable(
+            this IHealthChecksBuilder builder,
+            Uri endpoint,
+            TokenCredential tokenCredential,
+            string tableName,
+            string? name = default,
+            HealthStatus? failureStatus = default,
+            IEnumerable<string>? tags = default,
+            TimeSpan? timeout = default)
+        {
+            return builder.Add(new HealthCheckRegistration(
+               name ?? TABLE_NAME,
+               sp => new TableServiceHealthCheck(endpoint, tokenCredential, tableName),
+               failureStatus,
+               tags,
+               timeout));
+        }
+
+        /// <summary>
+        /// Add a health check for Azure CosmosDb/Azure Storage table.

```

Comment:
Azure CosmosDb or Azure Storage  ? Both?

Reply:
It's "or" so I can change it